railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -53,6 +53,8 @@ RUN curl -fsSL https://bun.sh/install | bash -s -- "bun-v${BUN_VERSION}"
 <% end -%>
 # Install application gems
 COPY Gemfile Gemfile.lock ./
+COPY vendor vendor
```

Comment:
It would be preferable to merge this with the COPY above.

Also the convention for vendored gems is the `vendor/bundle` directory.

Reply:
Merging this copy will copy vendor on every Gemfile change. This copy command does copy all subfolders so `vendor/bundle` should work. 

* I have tried to merge COPY commands w/o success. Closest to the docs will be `COPY Gemfile Gemfile.lock ./vendor .` but it fails on the Docker image build

Reply:
> This copy command does copy all subfolders so `vendor/bundle` should work.

The point is we shouldn't copy all of vendor.

Reply:
I see. There is a misunderstanding as I have used `vendored` word. So the issue is that some gems are not just unpacked but local, so they conventionally should live under `vendor/gems`. I can change the path to be that one. 
About 2 COPY commands - are you ok with them?

Reply:
My worry is that you may have lots of totally unrelated stuff in `vendor/`, e.g. JS etc.

> About 2 COPY commands - are you ok with them?

I'd prefer to limit the number of layer especially since the overwhelming majority of users won't have anything in `vendor/`

Reply:
Sure. I will do copy only of `vendor/gems`. Unfortunetely I'm sure there is no way to merge this copy instructions. Same as those 2 sit here: 
<img width="426" alt="image" src="https://github.com/user-attachments/assets/41bb2011-ebe5-4580-84d5-68d3b39295df">

https://github.com/rails/rails/blob/75794043032fa2cbf1d4c34841d997a11f2db401/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt#L101

Reply:
Conditional copy doesn't work for some reason. Anyway this folder will be copied to at last. So my proposed change will be - to copy whole vendor folder from within the app.

Reply:
I forced push your branch so merge the COPY.